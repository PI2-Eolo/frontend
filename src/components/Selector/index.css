.selector-container {
    display: flex;
    flex-direction: row;
    background-color: #eeecfa;
    border-radius: 1em;
    padding: 3px;
    width: fit-content;
}

button {
    display: flex;
    border: none;
    font-size: 16px;
    font-family: "Poppins-Thin";
    font-weight: 100;
    border-radius: 1em;
    padding: 5px;
    padding-right: 10px;
    padding-left: 10px;
}

button:hover {
    background-color: #033d68;
    color: #ffffff;
    transition-duration: 0.2s;
}

.unselected-button {
    background-color: #eeecfa;
    color: #033d68;
}

.selected-button {
    background-color: #033d68;
    color: #ffffff;
}

@font-face {
    font-family: "Poppins Thin";
    src: local("Poppins Thin"), url(../../assets/fonts/Poppins-Thin.ttf) format("ttf");
}