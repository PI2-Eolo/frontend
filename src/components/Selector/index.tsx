import './index.css';

interface SelectorProps {

}

const Selector = (props:SelectorProps) => {
    return (
        <div >
            Selector component
        </div>
    )
};

export default Selector;