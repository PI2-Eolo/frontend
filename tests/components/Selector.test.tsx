import renderer from "react-test-renderer";
import React from "react";
import Selector from "../../src/components/Selector";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
const selectorOptions = [
  {
    label: "Realtime",
    value: "realtime",
  },
  {
    label: "Daily",
    value: "daily",
  },
  {
    label: "Weekly",
    value: "weekly",
  },
  {
    label: "Monthly",
    value: "monthly",
  },
];

configure({adapter: new Adapter()});
describe("Selector component test", () => {
  it("match with snapshot", () => {
    const componentWithOptions = renderer.create(
      <Selector options={selectorOptions} />
    );
    expect(componentWithOptions.toJSON()).toMatchSnapshot();

    const componentWithDefaultValue = renderer.create(
      <Selector options={selectorOptions} defaultValue="daily" />
    );
    expect(componentWithDefaultValue.toJSON()).toMatchSnapshot();
  });

  it("Can call on change on button click", () => {
    const mockCallback = jest.fn();
    const selector = shallow(
      <Selector options={selectorOptions} onChange={mockCallback} />
    );
    expect(selector.find("button").length).toBe(4);
    selector.find("button").at(1).simulate("click");
    expect(mockCallback).toHaveBeenCalledWith("daily");
    selector.find("button").at(2).simulate("click");
    expect(mockCallback).toHaveBeenCalledWith("weekly");
  });
});
