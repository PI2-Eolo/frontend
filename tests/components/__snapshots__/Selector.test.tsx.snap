// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector component test match with snapshot 1`] = `
<div
  className="selector-container"
>
  <button
    className="selected-button"
    onClick={[Function]}
  >
    Realtime
  </button>
  <button
    className="unselected-button"
    onClick={[Function]}
  >
    Daily
  </button>
  <button
    className="unselected-button"
    onClick={[Function]}
  >
    Weekly
  </button>
  <button
    className="unselected-button"
    onClick={[Function]}
  >
    Monthly
  </button>
</div>
`;

exports[`Selector component test match with snapshot 2`] = `
<div
  className="selector-container"
>
  <button
    className="unselected-button"
    onClick={[Function]}
  >
    Realtime
  </button>
  <button
    className="selected-button"
    onClick={[Function]}
  >
    Daily
  </button>
  <button
    className="unselected-button"
    onClick={[Function]}
  >
    Weekly
  </button>
  <button
    className="unselected-button"
    onClick={[Function]}
  >
    Monthly
  </button>
</div>
`;
